include/slot_ranking.h

```diff
@@ -0,0 +1,153 @@
+/**
+ * @brief slot_ranking definition. slot_ranking contains the cosen action, along with the action probability distribution for a slot. 
+ */
+#pragma once
+#include "container_iterator.h"
+#include <cstddef>
+#include <iterator>
+#include <vector>
+#include <string>
+
+namespace reinforcement_learning {
+	class api_status;
+
+	/**
+	 * @brief Holds (action, probability) pairs, POD used for extern "C"
+	 */
+	struct action_prob_d {
+		//! action id
+		size_t action_id;
+		//! probability associated with the action id
+		float probability;
+	};
+
+	/**
+	 * @brief Holds (action, probability) pairs.
+	 */
+	struct action_prob : public action_prob_d {
+		inline action_prob(size_t action_id, float probability) {
+			this->action_id = action_id;
+			this->probability = probability;
+		};
+	};
+
+	/**
+	 * @brief
+	 * slot_ranking contains all the actions and distribution from with the action was sampled.  It also contains the chosen action id and
+	 * the unique join_id representing the choice.
+	 */
+	class slot_ranking {
+	public:
+		slot_ranking() = default;
+		~slot_ranking() = default;
+		/**
+			* @brief Construct a new slot detailed object.
+			*
+			* @param join_id The unique identifier for this interaction.  This event_id must be used when reporting the outcome for this action
+			*/
+		slot_ranking(char const* join_id);
+
+		/**
+		* @brief Unique join_id for this slot detailed instance.
+		* This join_id must be used when calling report_outcome so it can be joined with the chosen action
+		* @return const char*
+		*/
```

Comment:
join_id -> event_id? (same for the variable `_join_id`)
rlclientlib isn't really aware of joining afaik so it's a new term here.

Reply:
For ccb, all slots share an eventId, so the _slot_id uniquely identifies each slot for joining. The reason I introduced _join_id here is that slot_ranking is used in ranking_response, so I wanted to find a name that is applicable for both CB and CCB cases